biz.aQute.resolve/src/biz/aQute/resolve/MissingRequirementsException.java

```diff
@@ -0,0 +1,118 @@
+package biz.aQute.resolve;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.resource.Resource;
+import org.osgi.resource.Wire;
+import org.osgi.service.resolver.ResolutionException;
+
+import aQute.bnd.osgi.Processor;
+import aQute.bnd.osgi.resource.ResourceUtils;
+import aQute.bnd.osgi.resource.ResourceUtils.IdentityCapability;
+import aQute.lib.converter.Converter;
+import aQute.lib.exceptions.Exceptions;
+
+public class MissingRequirementsException extends ResolutionException {
+	private static final long serialVersionUID = 1L;
+
+	private final Map<Resource,List<Wire>>	wirings;
+	private final boolean					verbose;
+	private final boolean					trace	= false;
+
+	public MissingRequirementsException(Collection<Requirement> requirements,
+ Map<Resource,List<Wire>> wirings,
+			Processor properties) {
+		super((String) null, null, requirements);
+
+		this.wirings = wirings;
+
+		Boolean verbose;
+		try {
+			verbose = Converter.cnv(Boolean.class, properties.getProperty("-resolve.verbose", "true"));
+		} catch (Exception e) {
+			throw Exceptions.duck(e);
+		}
+
+		this.verbose = verbose.booleanValue();
+	}
+
+	@Override
+	public String getMessage() {
```

Comment:
We should probably first address the larger issue about whether MissingRequirementResource is the right approach.